src/app/subnets/views/SubnetDetails/StaticDHCPLease/ReserveDHCPLease/ReserveDHCPLease.tsx

```diff
@@ -77,65 +94,56 @@ const ReserveDHCPLease = ({
     }
   };
 
-  const initialValues = getInitialValues();
-
   const onClose = () => setSidePanelContent(null);
 
-  if (loading) {
-    return <Spinner text="Loading..." />;
-  }
-
-  if (!subnet) {
-    return null;
-  }
-
-  const [startIp, endIp] = getIpRangeFromCidr(subnet.cidr);
-  const [immutableOctets, _] = getImmutableAndEditableOctets(startIp, endIp);
-
   const ReserveDHCPLeaseSchema = Yup.object().shape({
     ip_address: Yup.string()
       .required("IP address is required")
       .test({
         name: "ip-is-valid",
         message: "This is not a valid IP address",
         test: (ip_address) => {
-          let valid = true;
-          const octets = `${ip_address}`.split(".");
-          octets.forEach((octet) => {
-            // IP address is not valid if the octet is not a number
-            if (isNaN(parseInt(octet))) {
-              valid = false;
-            } else {
-              const octetInt = parseInt(octet);
-              // Unsigned 8-bit integer cannot be less than 0 or greater than 255
-              if (octetInt < 0 || octetInt > 255) {
-                valid = false;
-              }
-            }
-          });
-          return valid;
+          if (subnetIsIpv4) {
+            return isIPv4(`${immutableOctets}.${ip_address}`);
+          } else {
+            return isIPv6(`${ipv6Prefix}${ip_address}`);
+          }
         },
       })
       .test({
         name: "ip-is-in-subnet",
         message: "The IP address is outside of the subnet's range.",
-        test: (ip_address) =>
-          isIpInSubnet(
-            `${immutableOctets}.${ip_address}`,
-            subnet?.cidr as string
-          ),
+        test: (ip_address) => {
+          if (subnetIsIpv4) {
+            return isIpInSubnet(
+              `${immutableOctets}.${ip_address}`,
+              subnet.cidr as string
+            );
+          } else {
+            try {
+              const addr = ipaddr.parse(`${ipv6Prefix}${ip_address}`);
+              const netAddr = ipaddr.parse(networkAddress);
+              return addr.match(netAddr, prefixLength);
+            } catch (e) {
+              return false;
+            }
+          }
+        },
       }),
     mac_address: Yup.string().matches(MAC_ADDRESS_REGEX, "Invalid MAC address"),
     comment: Yup.string(),
   });
 
   const handleSubmit = (values: FormValues) => {
+    const ip = subnetIsIpv4
+      ? `${immutableOctets}.${values.ip_address}`
+      : `${ipv6Prefix}${values.ip_address}`;
```

Comment:
The handleSubmit function could be simplified, e.g. by creating a helper function to format the IP address based on the type (IPv4 or IPv6).